### DISCUSSION TITLE
FAILURE: Build failed with an exception.

### DISCUSSION THREAD

**Discussion by tamharjeetwalia** [daniel-seiler/ImageTheming#6] [Q&A] [answered]
Get this error while using `gradle jar` to build a jar file. I'm not familiar with gradle and kotlin. I have searched for some similar errors and found that it might be the problem with the dependencies. But I have no idea. Is there something I did wrong?

Hope you might be able to help out 

![240128_20h39m42s_screenshot](https://github.com/daniel-seiler/ImageTheming/assets/127765756/eaab3bd0-832a-4644-97bf-043ba1dfedb2)

I also got the jar file from your [reddit post](https://www.reddit.com/r/unixporn/comments/pnc52k/oc_i_made_a_small_program_to_change_the_theme_of/) except the build is from the time the links weren't updated, because of which its not able to convert images. If i download the yml file and point the application to that manually it returns this error which i don't understand. Sorry 😔😔 

![240128_20h58m21s_screenshot](https://github.com/daniel-seiler/ImageTheming/assets/127765756/d46b30d7-27a0-464c-a412-10fed59d3071)

My java version:
openjdk 21.0.2 2024-01-16
OpenJDK Runtime Environment (build 21.0.2+13)
OpenJDK 64-Bit Server VM (build 21.0.2+13, mixed mode, sharing)

My Gradle version:
------------------------------------------------------------
Gradle 8.5
------------------------------------------------------------

Build time:   2023-11-29 15:40:31 UTC
Revision:     <unknown>

Kotlin:       1.9.20
Groovy:       3.0.17
Ant:          Apache Ant(TM) version 1.10.13 compiled on January 4 2023
JVM:          21.0.2 (Oracle Corporation 21.0.2+13)
OS:           Linux 6.7.1-arch1-1 amd64

**Comment by daniel-seiler** [comment, marked as answer]
Hey! It's nice to see people actually still using this project. A new version, which works with the new urls from Gogh, is available under releases or [here](https://github.com/daniel-seiler/ImageTheming/releases/tag/v0.4) (with added tilde support). That said, if you still want to build it yourself, use Gradle version 7.3 (or submit a pull request converting the gradle config to the newest version). The Project isn't compatible with the newer 8.x, hence the error. 

Regarding the error while running the application (remember to use the new .jar): remove the leading spaces in front of theme and path. Correct syntax would be: `java -jar ImageTheming.jar ~/Picture/wallpaper/whatever.jpg -t=~/Downloads/Atom.yml -o=~/Pictures/`

**Reply by tamharjeetwalia** [reply]
Oh my god, I'm actually hiding my face in embarrassment. How did I not check the releases section is beyond me. Thank you for helping me out and adding the tilde support. I don't have any experience with kotlin and gradle development so don't really know how to change the config, but learn some over the weekend or learn enough to be able submit a pull request.

Again, thank you Daniel for helping out
